File: socket_client.py
#coding=UTF-8
'''
测试TCP的慢启动,内容为<HTTP权威指南> P89 客户端
'''
from socket import *
from time import *
import traceback

def createClient():
	HOST=raw_input('send address')
	PORT=65535
	BUFSIZE=1024
	ADDR=(HOST,PORT)
	tcp=socket(AF_INET,SOCK_STREAM)
	tcp.connect(ADDR)
	data=[]
	try:
		print 'Wating for data'
		print 'starting send data:%s'%time()
		output=open("/tmp/big.data","r")
		buffer=[]
		count=0
		while True:
			data=output.read(BUFSIZE)
			if not data:
				break
			tcp.send(data)
		print 'ending  send data:%s'%time()
		output.close()
	except KeyboardInterrupt:
		tcp.close()
		exit()
	except Exception,e:
		print traceback.format_exc()
	#tcp.close()
if __name__ == '__main__':
	createClient()

File: socket_client.js
//测试TCP的慢启动,内容为<HTTP权威指南> P89 客户端
var net = require('net');
var fs=require('fs');
var socket_client=new net.Socket();
socket_client.setKeepAlive(true);
socket_client.connect(65535,"localhost",function(){
	console.log('start to send data info');
	fs.readFile('/tmp/big.data',function(err,data){
		socket_client.write(data);
	});
});
socket_client.on('close',function(){
	console.log('close!');
})